crates/chia-sdk-driver/src/layers/p2_eip712_message_layer.rs

```diff
@@ -0,0 +1,298 @@
+use chia_protocol::{Bytes, Bytes32};
+use clvm_traits::{FromClvm, ToClvm};
+use clvm_utils::{CurriedProgram, TreeHash};
+use clvmr::{Allocator, NodePtr};
+use ethers::utils::keccak256;
+use hex_literal::hex;
+
+use crate::{DriverError, Layer, Puzzle, Spend, SpendContext};
+
+pub const P2_EIP712_MESSAGE_PUZZLE: [u8; 276] = hex!("ff02ffff01ff02ffff03ffff22ffff20ffff8413d61f00ff17ff5fff81bf8080ffff20ffff24ffff01820ab9ffff0101ffff01ff02ffff03ffff09ffff3eff02ffff3eff05ff0bff178080ff2f80ff80ffff01ff088080ff0180ffff04ff05ffff04ff0bffff04ff2fffff04ffff02ff06ffff04ff02ffff04ff82017fff80808080ffff04ff5fff808080808080808080ffff01ff04ffff04ff04ffff04ff2fff808080ffff02ff82017fff8202ff8080ffff01ff088080ff0180ffff04ffff01ff46ff02ffff03ffff07ff0580ffff01ff0bffff0102ffff02ff06ffff04ff02ffff04ff09ff80808080ffff02ff06ffff04ff02ffff04ff0dff8080808080ffff01ff0bffff0101ff058080ff0180ff018080");
+pub const P2_EIP712_MESSAGE_PUZZLE_HASH: TreeHash = TreeHash::new(hex!(
+    "
+    aacce7b99db5b1e9eb16d676fa5f1a2e469ef589f29c4ab0010bac338a4df085
+    "
+));
+
+type EthPubkeyBytes = [u8; 33];
```

Comment:
This will be encoded as a list of 33 separate atoms. You want `BytesImpl<33>` and `BytesImpl<64>` instead of these aliases probably? Not sure why tests would pass here as is though.

Reply:
I had them as `Bytes` in the curried in args